internal/events/kafka.go

```diff
@@ -260,3 +276,44 @@ var _ io.Closer = (*KafkaConsumer)(nil)
 func (k *KafkaConsumer) Handlers() []EventHandler {
 	return k.handlers
 }
+
+// newTransportFromConfig creates new Kafka Transport from the given Kafka config
+func newTransportFromConfig(config KafkaConfig) (kafka.RoundTripper, error) {
+	var tlsConfig *tls.Config
+	transport := kafka.DefaultTransport
+	if slices.Contains(SSLProtocols, config.SecurityProtocol) {
+		tlsConfig = &tls.Config{}
+		if config.SecurityProtocol == KafkaProtocolSSL {
+			cert, err := tls.X509KeyPair([]byte(config.SSLAccessCertificate), []byte(config.SSLAccessKey))
+			if err != nil {
+				return nil, fmt.Errorf("parsing SSL access key and certificate: %w", err)
+			}
+			tlsConfig.Certificates = []tls.Certificate{cert}
+		}
+
+		transport = &kafka.Transport{
+			TLS: tlsConfig,
+		}
+	}
+
+	if config.SecurityProtocol == KafkaProtocolSASLPlaintext || config.SecurityProtocol == KafkaProtocolSASLSSL {
+		transport = &kafka.Transport{
+			SASL: plain.Mechanism{
+				Username: config.SASLUsername,
+				Password: config.SASLPassword,
+			},
+			TLS: tlsConfig,
+		}
+	}
+
+	return transport, nil
+}
+
+// closeOrLog closes the Kafka connection and logs the error if any
+func closeOrLog(conn *kafka.Conn) {
+	func(conn *kafka.Conn) {
+		if closeErr := conn.Close(); closeErr != nil {
+			log.Errorf("closing kafka connection: %v", closeErr)
+		}
+	}(conn)
+}
```

Comment:
Is there a reason for placing a function inside another function here?

Reply:
Thank you for catching that. It was a copy/paste mistake during refactor when extracting the method. 

addressed in d854b5f9e9ef197be9157b3b07c29e1a9f852fc7